includes/rest/class-outbox-controller.php

```diff
@@ -0,0 +1,302 @@
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The base of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = '(?:users|actors)/(?P<user_id>[\w\-\.]+)/outbox';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'user_id' => array(
+						'description'       => 'The ID of the user or actor.',
+						'type'              => 'string',
+						'validate_callback' => array( $this, 'validate_user_id' ),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( 'Activitypub\Rest\Server', 'verify_signature' ),
+					'args'                => array(
+						'page'     => array(
+							'description' => 'Current page of the collection.',
+							'type'        => 'integer',
+							'default'     => 1,
+							'minimum'     => 1,
+						),
+						'per_page' => array(
+							'description' => 'Maximum number of items to be returned in result set.',
+							'type'        => 'integer',
+							'default'     => 10,
+							'minimum'     => 1,
+							'maximum'     => 100,
+						),
+					),
+				),
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( 'Activitypub\Rest\Server', 'verify_signature' ),
+					'accept_json'         => true,
+				),
+				'schema' => array( $this, 'get_collection_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Validates the user_id parameter.
+	 *
+	 * @param mixed $user_id The user_id parameter.
+	 * @return bool|\WP_Error True if the user_id is valid, WP_Error otherwise.
+	 */
+	public function validate_user_id( $user_id ) {
+		$user = Actors::get_by_various( $user_id );
+		if ( \is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves a collection of outbox items.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return \WP_REST_Response|\WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		$user_id = $request->get_param( 'user_id' );
+		$page    = $request->get_param( 'page' );
+		$user    = Actors::get_by_various( $user_id );
+
+		/**
+		 * Action triggered prior to the ActivityPub profile being created and sent to the client.
+		 *
+		 * @param \WP_REST_Request $request The request object.
+		 */
+		\do_action( 'activitypub_rest_outbox_pre', $request );
+
+		$query_args = array(
+			'posts_per_page' => $request->get_param( 'per_page' ),
+			'author'         => $user_id > 0 ? $user_id : null,
+			'paged'          => $page,
+			'post_type'      => Outbox::POST_TYPE,
+			'post_status'    => 'draft',
+		);
+
+		$outbox_query = new \WP_Query();
```

Comment:
what about moving this to the outbox collection class? this way you can use it in different contexts?

Reply:
Let's wait until there's a need for it?